modules/manage/pages/kubernetes/security/authentication/k-user-controller.adoc

```diff
@@ -0,0 +1,245 @@
+
+You can specify the authentication type for a user using the `spec.authentication.type` field. Supported values include `scram-sha-256`, `scram-sha-512`, and their uppercase variants.
+
+[source,yaml]
+----
+spec:
+  authentication:
+    type: scram-sha-512
+----
+
+If no authentication credentials are provided, no user will be created, but ACLs can still be managed for existing users.
+
+=== Manage user secrets
+
+Redpanda users require a password, which you can provide directly, using the `spec.password.value` field, or through a Kubernetes Secret, using the `spec.password.valueFrom.secretKeyRef`. The Redpanda operator offers flexibility in how these secrets are handled:
+
+- If the Secret exists and the key exists within that Secret, the existing password will be used.
+- If the Secret exists but the key does not exist, the Secret will be updated with an autogenerated password.
+- If the Secret does not exist, a new Secret with the provided key will be created with an autogenerated password.
+
+This behavior ensures that you can manage user credentials securely and programmatically, while also automating password generation if necessary.
+
+To use an existing Kubernetes Secret, ensure that the Secret and key are both defined. For example:
+
+[source,yaml]
+----
+spec:
+  authentication:
+    password:
+      valueFrom:
+        secretKeyRef:
+          name: user-secret
+          key: password
+----
+
+This example assumes that a Kubernetes Secret named `user-secret` with a key `password` exists. If the Secret does not exist or the key is missing, the Redpanda Operator will handle it by creating or updating the Secret with an autogenerated password. The autogenerated password will follow best practices for secure password generation.
+
+If you need to create a Secret, you can use the following command as an example:
+
+[source,bash]
+----
+kubectl --namespace <namespace> create secret generic user-secret --from-file=password.txt
+----
+
+In this example, the `password.txt` file contains the password you want to use.
+
+=== Define ACLs
+
+The `spec.authorization` field allows you to manage ACLs for users. ACLs define the permissions users have over specific resources in Redpanda, such as topics, consumer groups, and clusters.
+
+You can define ACLs for a user by specifying which resources they can access and the operations they are permitted to perform. Here's an example configuration for managing ACLs:
+
+[source,yaml]
+----
+spec:
+  authorization:
+    acls:
+      - type: allow
+        resource:
+          type: topic
+          name: my-topic
+          patternType: literal
+        operations: [Read, Write]
+----
+
+- `type`: Defines whether the ACL is `allow` or `deny`.
+- `resource.type`: Specifies the resource type.
+- `patternType`: Specifies if the resource name is treated as a `literal` or a `prefixed` pattern.
+- `operations`: Lists the allowed operations, such as `Read`, `Write`, `Create`, and `Delete`.
```

Comment:
Might be worth linking to the CRD docs for all of the supported ops and resource types?

Reply:
Gah, just realized that the parser doesn't work great for enum types (it'd likely need to know how to parse the enum validation option [here](https://github.com/redpanda-data/redpanda-operator/blob/3b3a40b4d9f55d10e726e97a32230a175196469c/operator/api/redpanda/v1alpha2/user_types.go#L378)), so I'll probably need to add in some docs inline on the CRD itself to show the supported types:

https://deploy-preview-773--redpanda-docs-preview.netlify.app/24.3/reference/k-crd/#k8s-api-github-com-redpanda-data-redpanda-operator-src-go-k8s-api-redpanda-v1alpha2-resourcetype

Reply:
I regenerated the CRD reference with the new descriptions.